test/e2e/reconciler/cache/replication_test.go

```diff
@@ -60,109 +61,264 @@ type testScenario struct {
 		return true, ""
 	}, wait.ForeverTestTimeout, 400*time.Millisecond)
-	t.Logf("Get root/%s/%s APIExport from the cache server", cluster, "wild.wild.west")
+	t.Logf("Get %s/%s/%s from the cache server", scenario.resourceKind, cluster, scenario.resourceName)
 	framework.Eventually(t, func() (bool, string) {
-		cachedWildAPIExport, err := cacheKcpClusterClient.Cluster(cluster).ApisV1alpha1().APIExports().Get(cacheclient.WithShardInContext(ctx, shard.New("root")), wildAPIExport.Name, metav1.GetOptions{})
+		originalResource, err := scenario.getSourceResource(cluster)
+		if err != nil {
+			return false, err.Error()
+		}
+		cachedResource, err := scenario.getCachedResource(cluster)
+		if err != nil {
+			return false, err.Error()
+		}
+		t.Logf("Verify if both the orignal and replicated resources (%s/%s/%s) are the same except %s annotation and ResourceVersion after modification", scenario.resourceKind, cluster, scenario.resourceName, genericapirequest.AnnotationKey)
+		cachedResourceMeta, err := meta.Accessor(cachedResource)
 		if err != nil {
 			return false, err.Error()
 		}
-		t.Logf("Verify if both the orignal APIExport and replicated are the same except %s annotation and ResourceVersion after an update to the spec", genericapirequest.AnnotationKey)
-		if _, found := cachedWildAPIExport.Annotations[genericapirequest.AnnotationKey]; !found {
-			return false, fmt.Sprintf("replicated APIExport root/%s/%s, doesn't have %s annotation", cluster, cachedWildAPIExport.Name, genericapirequest.AnnotationKey)
+		if _, found := cachedResourceMeta.GetAnnotations()[genericapirequest.AnnotationKey]; !found {
+			t.Fatalf("replicated %s root/%s/%s, doesn't have %s annotation", scenario.resourceKind, cluster, cachedResourceMeta.GetName(), genericapirequest.AnnotationKey)
 		}
-		delete(cachedWildAPIExport.Annotations, genericapirequest.AnnotationKey)
-		if diff := cmp.Diff(cachedWildAPIExport, updatedWildAPIExport, cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion")); len(diff) > 0 {
-			return false, fmt.Sprintf("replicated APIExport root/%s/%s is different that the original, diff: %s", cluster, wildAPIExport.Name, diff)
+		delete(cachedResourceMeta.GetAnnotations(), genericapirequest.AnnotationKey)
+		if diff := cmp.Diff(cachedResource, originalResource, cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion")); len(diff) > 0 {
+			return false, fmt.Sprintf("replicated %s root/%s/%s is different that the original", scenario.resourceKind, cluster, cachedResourceMeta.GetName())
 		}
 		return true, ""
 	}, wait.ForeverTestTimeout, 400*time.Millisecond)
 }
 
-// TestAllAgainstInProcessCacheServer runs all test scenarios against a cache server that runs with a kcp server
-func TestAllAgainstInProcessCacheServer(t *testing.T) {
+// TestAllReplicationScenariosAgainstInProcessCacheServer runs all test scenarios against a cache server that runs with a kcp server
+func TestAllReplicationScenariosAgainstInProcessCacheServer(t *testing.T) {
 	t.Parallel()
```

Comment:
This makes the suite parallel with the rest of the suites, but does not make each test case parallel - is that intentional? Do we want a `t.Parallel()` call in the `t.Run()`?

Reply:
I don't think so, too much concurrency can negatively impact a CPU. 
There is no difference on my local machine: `53`s vs `52`s